lib/log-parser.js

```diff
@@ -0,0 +1,76 @@
+
+import fs from 'fs'
+import readline from 'readline'
+import LogEntry from 'log-entry'
+import moment from 'moment'
+
+const startDatePattern = /^Started '(.*)' on(.*)$/
+const logEntryPattern = /^\[(\d{2}:\d{2}:\d{2})\][ |\w]:(\t*)(.*)$/
+
+const parseStartDate = (lineWithDate) => {
+  const result = lineWithDate.match(startDatePattern)
+  if (result) {
+    return moment(result[1], 'ddd MMM DD HH:mm:ss PST yyyy').toDate()
+  }
+  return null
+}
+
+const parseLogDate = (logDateString, startDate) => {
+  const logDate = new Date(startDate.getTime())
+
+  const hour = logDateString.substring(0, 2)
+  const min = logDateString.substring(3, 5)
+  const sec = logDateString.substring(6, 8)
+
+  logDate.setHours(hour, min, sec, 0)
+
+  if (logDate < startDate) {
+    //if log finieshed next day
+    logDate.setDate(logDate.getDate() + 1)
+  }
+
+  return logDate
+}
+
+const parseLine = (logLine, startDate) => {
+  const result = logLine.match(logEntryPattern)

```

Comment:
name of the variable is meaningless, result of what? It's actually an array of matches, so `logEntryMatches` is far better name